lib/xapian_db/index_writers/sidekiq_writer.rb

```diff
@@ -15,39 +15,45 @@ class SidekiqWriter
       end
 
       class << self
-
-        # Update an object in the index
-        # @param [Object] obj An instance of a class with a blueprint configuration
-
         def queue
           XapianDb::Config.sidekiq_queue
         end
 
+        # Update an object in the index
+        # @param [Object] obj An instance of a class with a blueprint configuration
         def index(obj, _commit= true, changed_attrs: [])
-          Sidekiq::Client.enqueue_to(queue, worker_class, 'index',
-                                     {
-                                       class: obj.class.name,
-                                       id: obj.id,
-                                       changed_attrs: changed_attrs
-                                     }.to_json)
+          Sidekiq::Client.push('queue' => queue,
```

Comment:
Both `Sidekiq::Client.push` and `Sidekiq::Client.enqueue_to` methods are used to enqueue jobs in Sidekiq. The difference is:
- The [Sidekiq::Client.enqueue_to](https://github.com/sidekiq/sidekiq/blob/main/lib/sidekiq/client.rb#L196) method provides a higher-level, more straightforward way to enqueue a job to a specific queue. It is useful when you just need to specify the `queue`, `worker class`, and `arguments`, without worrying about additional job options.
- The [Sidekiq::Client.push](https://github.com/sidekiq/sidekiq/blob/ed8b217d32f026acc1fae7221dca18d1baa86d11/lib/sidekiq/client.rb#L86) method provides a low-level interface for enqueuing jobs. It allows you to define a complete set of job attributes, giving you full control over the job's configuration, including the `queue`, `worker class`, `arguments`, `retry` behavior, and other options.

As we need to pass `retry` option and `enqueue_to` method doesn't accept `retry`, hence,  we must use `Sidekiq::Client.push` method.

`N.B.` I have tested locally changing the retry value, it works fine 👍 .